colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -160,7 +160,6 @@ class ServiceClient[P <: Protocol](
     val context: Context
 )(implicit tagDecorator: TagDecorator[P] = TagDecorator.default[P])
     extends ControllerDownstream[Encoding.Client[P]]
-    with HasUpstream[ControllerUpstream[Encoding.Client[P]]]
```

Comment:
https://github.com/tumblr/colossus/blob/master/colossus/src/main/scala/colossus/controller/Controller.scala#L43
`trait ControllerDownstream[E <: Encoding] extends HasUpstream[ControllerUpstream[E]]` so it's redundant